vars/buildImage.groovy

```diff
@@ -0,0 +1,99 @@
+// Create a Job and BuildConfig Template to trigger the Build
+// Available parameters:
+//    dockerFile    string -- DockerFile used for the buildconfig
+//    gitURL:       string -- Git URL used for the buildconfig
+//    image         string -- Image used for the buildconfig
+//    nameSpace:    string -- Namespace name to run the build
+def call(params = [:]) {
+    def imageName
+    pipeline {
+        // We don't need any specify agent just to add the buildconfig and trigger it,
+        // this pod will be destroyed in the end
+        agent {
+            node { label 'nodejs' }
+        }
+        stages {
+            stage('Prepare Env') {
+                steps{
+                   script{
+                       openshift.withCluster() {
+                           openshift.withProject() {
+
+                               def bcJSON  = libraryResource 'com/github/coreos/buildconfig.json'
+                               def jobJSON = libraryResource 'com/github/coreos/job.json'
+                               def jobObj  = readJSON text: jobJSON
+                               def bcObj   = readJSON text: bcJSON
+                               def tagName = UUID.randomUUID()
+
+                               jobObj['parameters'][0] +=['name': 'NAME', 'value': "cleanup-${tagName}".toString()]
```

Comment:
Instead of setting parameter values like this, we can do like described here: https://github.com/openshift/jenkins-client-plugin#cannot-live-without-openshift-templates-no-problem and use `oc.process(obj, "-p", "NAME=...")`.

Reply:
I was able to do it for the job template. But for the buildconfig even though I tried the same steps,  some reason the buildoconfig and the imagestream are not created at the end. There isn't any error ou logs to see,  I can only see the parameters is changed ok. Not sure why, probably due the way this template was created, I spent some time on that but couldn't figure out.  I will need to dig more to understand why I can't due it in the second case